DotNetCoreMasters/DotNetCoreMasters/Filter/ShowElapseTimeAttribute.cs

```diff
@@ -0,0 +1,35 @@
+﻿using Microsoft.AspNetCore.Mvc.Filters;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace DotNetCoreMasters.Filter
+{
+    public class ShowElapseTimeAttribute : IResultFilter
```

Comment:
Please use a Resource filter instead in order to capture the elapsed time on the MVC filter pipeline request.

`IResultFilter` or any other filter types will behave as a singleton by default so there could be race condition problems when multiple requests are received at the same time. You should create an instance of stopwatch per request and store them in the HttpContext. The way to store is indicated on the Session 7 PPT page 60.